examples/api-tests/src/scm.spec.js

```diff
@@ -0,0 +1,153 @@
+/********************************************************************************
+ * Copyright (C) 2020 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+// @ts-check
+describe('SCM', function () {
+
+    const { assert } = chai;
+
+    const Uri = require('@theia/core/lib/common/uri');
+    const { ApplicationShell } = require('@theia/core/lib/browser/shell/application-shell');
+    const { ScmContribution } = require('@theia/scm/lib/browser/scm-contribution');
+    const { ScmService } = require('@theia/scm/lib/browser/scm-service');
+    const { ScmWidget } = require('@theia/scm/lib/browser/scm-widget');
+
+    /** @type {import('inversify').Container} */
+    const container = window['theia'].container;
+    const scmContribution = container.get(ScmContribution);
+    const shell = container.get(ApplicationShell);
+
+    /** @type {ScmWidget} */
+    let scmWidget;
+
+    /** @type {ScmService} */
+    let scmService;
+
+    beforeEach(async () => {
+        await shell.leftPanelHandler.collapse();
+        scmWidget = await scmContribution.openView({ activate: true, reveal: true });
+        scmService = scmWidget['scmService'];
+    });
+
+    describe('scm-view', () => {
+
+        it('the view should open and activate successfully', () => {
+            assert.notEqual(scmWidget, undefined);
+            assert.strictEqual(scmWidget, shell.activeWidget);
+        });
+
+        describe('\'ScmTreeWidget\'', () => {
+
+            it('the view should display the resource tree when a repository is present', () => {
+                const resourceWidget = scmWidget['resourceWidget'];
```

Comment:
we should rather replace private index with proper API, i.e. add public getter. If you need them to write tests, it means that downstream will find them useful too.